### PR TITLE
feat: Graduate EKS out of experimental

### LINKED ISSUE #2539: AWSClusterStaticIdentity requires duplicate secret for EKS
/kind bug
/area ux
/area security
/help
/priority important-soon

**What steps did you take and what happened:**
When using `AWSClusterStaticIdentity` with `AWSManagedControlPlane` and machines, it requires that the secret that is referenced by `AWSClusterStaticIdentity` be in 2 different namespaces:

- capa-system (for the main infra controller manager)
- capa-eks-control-plane-system (for the control plane provider)

This is because the credentials are used in 2 different controller managers that reside in different namespaces, and in the code we are using [system.GetManagerNamespace](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f3ce37917aafd6f8d351340846b37cdd42ecf7ce/pkg/cloud/scope/session.go#L343).

This results in the secret having to be duplicated.

As an example in the [EKS e2e tests](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/shared/identity.go) we duplicate the secret.

**What did you expect to happen:**
I would expect to only create the secret once.

**Anything else you would like to add:**
There are a couple of ways to fix this:
1) Change `buildAWSClusterStaticIdentity` to look for the secret in a specific namespace (probably just the ns of the main infra manager, so capa-system)
2) We revisit having a separate control plane provider and move its controllers into the main infra provider. This would solve this issue and also make clusterctl simpler.

**Environment:**

- Cluster-api-provider-aws version: 0.6.6

### LINKED ISSUE #2570: Graduate EKS out of experimental
/kind refactor
/area provider/eks
/priority important-soon
/milestone v0.7.0

**Describe the solution you'd like**
The EKS functionality first introduced in 0.6.0 (9th September 2020) is currently experimental. This issue proposes that we graduate the EKS functionality out of experimental. Specifically, the following would be graduated:

- AWSManagedControlPlane
- EKSConfig

The AWSManagedMachinePool will stay experimental as MachinePool is still experimental.

**Anything else you would like to add:**
This will be done as part of merging the controplane & bootstrap provider into the main infra provider

**Environment:**

- Cluster-api-provider-aws version: 0.6.6

### PR DESCRIPTION
**Pull request by richardcase**
Signed-off-by: Richard Case <richard@weave.works>

**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

The EKS functionality is graduated out of experimental with this change. This means it will be enabled by default. 

As part of the graduation the separate controlplane and bootstrap providers have been removed and
merged into the main AWS infrastructure provider.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2570
Fixes #2539 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
action required
With the graduation of EKS out of experimental and it being enabled by default **all** users of Cluster API Provider AWS need to be aware of the following:

1. EKS support is now enabled by default. If you don't want to enable the EKS functionality then you will need to disable the creation of the IAM permissions via a `clusterawsadm` configuration file:

apiVersion: bootstrap.aws.infrastructure.cluster.x-k8s.io/v1alpha1
kind: AWSIAMConfiguration
spec:
  eks:
    disable: true

clusterawsadm bootstrap iam create-cloudformation-stack --config bootstrap-config.yaml

and then you need to set the `CAPA_EKS` environment variable to false before doing `clusterctl init`. For example:

export CAPA_EKS=false
clusterctl init --infrastructure=aws

2.  The EKS bootstrap and controlplane providers have been merged into the main infrastructure provider/manager. You will need to remove any reference to aws-eks in the control-plane/bootstrap flags for clusterctl init:

For example, change this:

clusterctl init --infrastructure=aws --control-plane aws-eks --bootstrap aws-eks

To this:

clusterctl init --infrastructure=aws

3. AWSManagedCluster has been removed. If you are using AWSManagedCluster then you will need to replace it with a reference to the AWSmanagedControlPlane. For example, change this:
apiVersion: cluster.x-k8s.io/v1alpha4
kind: Cluster
metadata:
  name: "cl1"
spec:
  clusterNetwork:
    pods:
      cidrBlocks: ["192.168.0.0/16"]
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
    kind: AWSManagedCluster
    name: "cl1"
  controlPlaneRef:
    kind: AWSManagedControlPlane
    apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
    name: "cl1-control-plane"

to this:

apiVersion: cluster.x-k8s.io/v1alpha4
kind: Cluster
metadata:
  name: "cl1"
spec:
  clusterNetwork:
    pods:
      cidrBlocks: ["192.168.0.0/16"]
  infrastructureRef:
    kind: AWSManagedControlPlane
    apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
    name: "cl1-control-plane"
  controlPlaneRef:
    kind: AWSManagedControlPlane
    apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
    name: "cl1-control-plane"

4. Fargate profile support remains experimental and as suck must be enabled using the **EKSFargate** feature flag. This can be done before running `clusterctl init` using the **EXP_EKS_FARGATE** environment variable:

export EXP_EKS_FARGATE=true
clusterctl init --infrastructure=aws

You will also need to ensure you have the fargate default profile created. See the docs for further information.

```

### CODE REVIEW

**Review by sedefsavas — COMMENTED** [review]
I think we also need to update the quick start (in core cluster-api repo) and disable EKS permissions by passing a config to clusterawsadm there.

**Comment by richardcase** [comment]
> I think we also need to update the quick start (in core cluster-api repo) and disable EKS permissions by passing a config to clusterawsadm there.

I have added a section on this in the disabling eks documentation topic

File: docs/book/src/topics/using-clusterawsadm-to-fulfill-prerequisites.md [hunk 1]
```diff
@@ -71,19 +71,18 @@ These will be added to the control plane and node roles respectively when they a
 
 #### With EKS Support
```
[the remark this answers is not in the document]
Comment (richardcase) [reply docs/book/src/topics/using-clusterawsadm-to-fulfill-prerequisites.md:72]: Ah yes, thats a great spot, thank you. I will update now

**Comment by sedefsavas** [comment]
> > I think we also need to update the quick start (in core cluster-api repo) and disable EKS permissions by passing a config to clusterawsadm there.
> 
> I have added a section on this in the disabling eks documentation topic

Yes, but for users that will not use EKS support, they will not read through the EKS section. In the quick start, we are not bringing up EKS clusters so the permissions for it is not needed.

**Comment by richardcase** [comment]
> In the quick start, we are not bringing up EKS clusters so the permissions for it is not needed.

I need to make a change to clusterctl to remove the bootstrap/controlplane providers so i will add a note to the quick start starts about disabling EKS support. Would that be ok?

File: docs/book/src/topics/using-clusterawsadm-to-fulfill-prerequisites.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardcase) [reply docs/book/src/topics/using-clusterawsadm-to-fulfill-prerequisites.md:72]: Updated the developers guide

**Comment by richardcase** [comment]
Until the EKS e2e have passed multiple times:

/hold

**Comment by randomvariable** [comment]
All looks reasonable to me.
Going to give this a
/lgtm

**Comment by richardcase** [comment]
The e2e tests [passed](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/2648/pull-cluster-api-provider-aws-e2e-eks/1425364530233348096) for a third time.

**Comment by richardcase** [comment]
@randomvariable - sorry i had to resolve a conflict so the lgtm has done.

**Comment by randomvariable** [comment]
/approve

Remove /hold wehn you're comfortable @richardcase

**Comment by richardcase** [comment]
> Remove /hold wehn you're comfortable @richardcase
Will do, thanks @randomvariable

**Comment by richardcase** [comment]
After a flake in the teardown another pass of the [e2e](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/2648/pull-cluster-api-provider-aws-e2e-eks/1425900060347994112)

**Comment by richardcase** [comment]
//test pull-cluster-api-provider-aws-e2e

**Comment by richardcase** [comment]
Unmanaged e2e are [passing](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/2648/pull-cluster-api-provider-aws-e2e/1425927558221795328) as well.

So its time to:

/unhold

**Pull request merged**